app/lib/tool/test_profile/importer.dart

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:gcloud/db.dart';
+import 'package:gcloud/service_scope.dart';
+import 'package:http/http.dart';
+import 'package:meta/meta.dart';
+import 'package:path/path.dart' as p;
+
+import 'package:client_data/package_api.dart';
+
+import '../../account/backend.dart';
+import '../../account/models.dart';
+import '../../package/backend.dart';
+import '../../publisher/models.dart';
+import '../../shared/urls.dart' as urls;
+
+import 'models.dart';
+import 'resolver.dart';
+
+/// Imports [profile] data into the Datastore.
+@visibleForTesting
+Future<void> importProfile({
+  @required TestProfile profile,
+  List<String> resolvedVersions,
+  @required String archiveCachePath,
+}) async {
+  // resolve versions if they are not yet resolved
+  if (resolvedVersions == null || resolvedVersions.isEmpty) {
+    resolvedVersions = await resolveVersions(profile);
+  }
+
+  // create users
+  for (final u in profile.users) {
+    final oauthUserId = _baseIdFromEmail(u.email);
+    final userId = _userIdFromEmail(u.email);
+    await dbService.commit(inserts: [
+      User()
+        ..id = userId
+        ..oauthUserId = oauthUserId
+        ..email = u.email
+        ..created = DateTime.now().toUtc()
+        ..isDeleted = false,
+      OAuthUserID()
+        ..id = oauthUserId
+        ..userIdKey = dbService.emptyKey.append(User, id: userId),
+    ]);
+  }
+
+  // create publishers
+  for (final p in profile.publishers) {
+    await dbService.commit(inserts: [
+      Publisher()
+        ..id = p.name
+        ..contactEmail = p.members.isEmpty ? null : p.members.first.email
+        ..websiteUrl = 'https://${p.name}/'
+        ..created = DateTime.now().toUtc()
+        ..updated = DateTime.now().toUtc()
+        ..isAbandoned = false,
+      ...p.members.map(
+        (m) => PublisherMember()
+          ..parentKey = dbService.emptyKey.append(Publisher, id: p.name)
+          ..id = _userIdFromEmail(m.email)
+          ..userId = _userIdFromEmail(m.email)
+          ..created = DateTime.now().toUtc()
+          ..updated = DateTime.now().toUtc()
+          ..role = PublisherMemberRole.admin,
+      )
+    ]);
+  }
+
+  // create packages
+  Client client;
+  final archiveCacheDir = Directory(archiveCachePath);
+  await archiveCacheDir.create(recursive: true);
+  for (final rv in resolvedVersions) {
+    final parts = rv.split(':');
```

Comment:
You might want to use: https://pub.dev/documentation/basics/latest/lib_string_basics/StringBasics/partition.html

So that we only split in two...